### PR TITLE
[Auditor] Add more ignored libraries

### PR DESCRIPTION
**Pull request by giordano**
Our audit logs contain lots of references to system libraries.  I've been
meaning to do this for a long time, I'll probably keep this open for some more
time to collect more libraries we spot into the wild, but I wanted to open this
PR to make sure this is a good idea.  CC: @staticfloat

### CODE REVIEW

File: src/auditor/dynamic_linkage.jl [hunk 1]
```diff
@@ -184,18 +184,34 @@ function should_ignore_lib(lib, ::ELFHandle)
         "libthr.so.3",
         "libpthread.so.0",
     ]
+    if Sys.isfreebsd(platform)
+        push!(ignore_libs,
+              # From FreeBSD SDK
+              "libdevstat.sos.7",
+              "libexecinfo.so.1",
+              "libkvm.so.7",
+              "libz.so.6",
```
Comment (staticfloat) [inline src/auditor/dynamic_linkage.jl:193]: I'm not 100% sure this is a good idea, as we may want to ensure that `libz.so.6` comes from our Zlib_jll` sometimes?

File: src/auditor/dynamic_linkage.jl [hunk 2]
```diff
@@ -184,18 +184,34 @@ function should_ignore_lib(lib, ::ELFHandle)
...
+function should_ignore_lib(lib, ::MachOHandle, platform::AbstractPlatform)
     ignore_libs = [
         "libsystem.b.dylib",
         # This is not built by clang or GCC, so we leave it as a system library
         "libc++.1.dylib",
         "libresolv.9.dylib",
+        # Other libraries in the SDK
+        "libz.1.dylib",
+        # Frameworks in the SDK
+        "courefoundation",
```
Comment (jeremiahpslewis) [inline src/auditor/dynamic_linkage.jl:207]:
```suggestion
        "corefoundation",
```

**Review by jeremiahpslewis — COMMENTED** [review]
Just leafing through the build_tarball.jl files for things that have to get frequently rm'd, what about:


```bash
libexpat.so.*
libcrypto.so*
libssl.so*
```

**Comment by giordano** [comment]
> Just leafing through the build_tarball.jl files for things that have to get frequently rm'd, what about:
> 
> 
> ```bash
> libexpat.so.*
> libcrypto.so*
> libssl.so*
> ```

That goes precisely in the opposite direction of what Elliot said above

**Comment by giordano** [comment]
Ok, I'm going ahead and merge this.  It already silences plenty of warnings, for example all 52 warnings in [this recent PR](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=17188&view=results).  Warnings in CI of BinaryBuilder were almost halved, from 28 to 16

**Pull request merged**